src/wrap_fn_ptr.rs

```diff
@@ -1,37 +1,23 @@
-/// A `trait` to extract the `fn` ptr type of a [`WrappedFnPtr`].
-pub trait HasFnPtr {
-    type FnPtr;
-}
-
-/// A newtype wrapped `fn` ptr.
-///
-/// This allows us to add a safer (type-safe for sure, and increasingly fully safe)
-/// wrapper around calling a `fn` ptr.
-#[repr(transparent)]
-pub struct WrappedFnPtr<F>(F);
-
-impl<F> HasFnPtr for WrappedFnPtr<F> {
-    type FnPtr = F;
-}
-
-impl<F> WrappedFnPtr<F> {
-    pub const fn new(fn_ptr: F) -> Self {
-        Self(fn_ptr)
-    }
-
-    pub const fn get(&self) -> &F {
-        &self.0
-    }
-}
-
 /// This declares a wrapper for a `fn` ptr
 /// and defines related, useful things for that `fn` ptr.
 ///
 /// The API for [`wrap_fn_ptr!`] is a `fn` signature with no body.
 /// This generates a `mod` with the name of the `fn` provided that contains:
 ///
+/// * `type FnPtr`:
+///     The raw, inner `fn` ptr (according to the provided signature) contained by `Fn`.
+///
 /// * `type Fn`:
-///     A [`WrappedFnPtr`] wrapping the `fn` ptr signature provided.
+///     A newtype wrapping `FnPtr`.
+///     It defines `const fn new(FnPtr) -> Self` to construct it
+///     and `const fn get(&self) -> &FnPtr` to read the `FnPtr`.
+///
+///     These methods are marked `pub(super)`
+///     as they are meant to be used in the module calling [`wrap_fn_ptr!`].
+///
+///     It is meant for a `fn call` to also be added
+///     to call the `fn` in a type-safe (e.x. [`BitDepth`]-wise)
```

Comment:
```suggestion
///     It is meant for a `fn call` method to also be implemented
///     for this type to allow users to call the `fn` in a type-safe
///     (e.x. [`BitDepth`]-wise)
```

Reply:
Clarifies a couple things that took me a minute to understand, does this look ok?

Reply:
Yeah, that sounds clearer.

Reply:
Fixed in 1816aaa.